compiler/rustc_expand/src/mbe/macro_check.rs

```diff
@@ -261,7 +261,17 @@ fn check_binders(
             }
         }
         // Similarly, this can only happen when checking a toplevel macro.
-        TokenTree::MetaVarDecl(span, name, _kind) => {
+        TokenTree::MetaVarDecl(span, name, kind) => {
+            if kind.is_none() && node_id != DUMMY_NODE_ID {
+                // FIXME: Report this as a hard error eventually and remove
+                // equivalent errors from `parse_tt_inner` and `nameize`.
```

Comment:
It's worth mentioning that this setup causes these errors to be emitted twice; once as a hard error, and once as a lint error.

Reply:
I wonder if it's worth mentioning in the hard error that it will be duplicated with a lint error (if the lint isn't ignored). Just to pre-empt the inevitable bug reports about the error occurring twice. Not sure.